contracts/examples/SimpleW3box.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "./FlatDirectory.sol";
+
+contract SimpleW3box {
+    using Strings for uint256;
+
+    // event for EVM logging
+    event OwnerSet(address indexed oldOwner, address indexed newOwner);
+
+    // modifier to check if caller is owner
+    modifier isOwner() {
+        // If the first argument of 'require' evaluates to 'false', execution terminates and all
+        // changes to the state and to Ether balances are reverted.
+        // This used to consume all gas in old EVM versions, but not anymore.
+        // It is often a good idea to use 'require' to check if functions are called correctly.
+        // As a second argument, you can also provide an explanation about what went wrong.
+        require(msg.sender == owner, "Caller is not owner");
+        _;
+    }
+
+    struct File {
+        uint256 time;
+        bytes name;
+        bytes fileType;
+    }
+
+    struct FilesInfo {
+        uint256 length;
+        File[] files;
+        mapping(bytes32 => uint256) fileIds;
+    }
+
+    FlatDirectory public fileFD;
+
+    address public owner;
+    string public gateway;
+
+    mapping(address => FilesInfo) fileInfos;
+
+    constructor(string memory _gateway) {
+        owner = msg.sender;
+        gateway = _gateway;
+        fileFD = new FlatDirectory(0);
+    }
+
+    receive() external payable {
+    }
+
+    function setGateway(string calldata _gateway) public isOwner {
+        gateway = _gateway;
+    }
+
+    function write(bytes memory name, bytes memory fileType, bytes calldata data) public payable {
+        writeChunk(name, fileType, 0, data);
+    }
+
+    function writeChunk(bytes memory name, bytes memory fileType, uint256 chunkId, bytes calldata data) public payable {
+        bytes32 nameHash = keccak256(name);
+        FilesInfo storage info = fileInfos[msg.sender];
+        if (info.fileIds[nameHash] == 0) {
+            // first add file
+            info.files.push(File(block.timestamp, name, fileType));
+            info.fileIds[nameHash] = info.files.length;
+            info.length++;
+        }
+
+        fileFD.writeChunk{value: msg.value}(getNewName(msg.sender, name), chunkId, data);
+    }
+
+    function remove(bytes memory name) public returns (uint256) {
+        bytes32 nameHash = keccak256(name);
+        FilesInfo storage info = fileInfos[msg.sender];
+        require(info.fileIds[nameHash] != 0, "File does not exist");
+
+        delete info.files[info.fileIds[nameHash] - 1];
```

Comment:
swap the last obj with the to-delete one and pop() will be better solution. And you may don't need the length var